src/ImageSharp/Formats/Jpeg/JpegEncoderCore.cs

```diff
@@ -167,6 +171,54 @@ private void WriteJfifApplicationHeader(ImageMetadata meta, Span<byte> buffer)
         this.outputStream.Write(buffer, 0, 18);
     }
 
+    /// <summary>
+    /// Writes comment
+    /// </summary>
+    /// <param name="metadata">The image metadata.</param>
+    private void WriteComment(JpegMetadata metadata)
+    {
+        int maxCommentLength = 65533;
+
+        if (metadata.Comments.Count == 0)
+        {
+            return;
+        }
+
+        for (int i = 0; i < metadata.Comments.Count; i++)
+        {
+            string comment = metadata.Comments[i].ToString();
+
+            if (comment.Length > maxCommentLength)
+            {
+                string splitComment = comment.Substring(maxCommentLength, comment.Length - maxCommentLength);
+                metadata.Comments.Insert(i + 1, JpegComData.FromString(splitComment));
+
+                // We don't want to keep the extra bytes
+                comment = comment.Substring(0, maxCommentLength);
+            }
+
+            int commentLength = comment.Length + 4;
+
+            Span<byte> commentSpan = new byte[commentLength];
```

Comment:
You should be able to avoid the allocation here by using a single buffer of the maximum length plus markers outside of the loop and slicing.